meson.build

```diff
@@ -1332,6 +1332,12 @@ libopenssl = dependency('openssl',
                         version : '>= 1.1.0',
                         required : get_option('openssl'))
 conf.set10('HAVE_OPENSSL', libopenssl.found())
+if libopenssl.found()
+        have = cc.has_header('openssl/ui.h') and cc.has_function('UI_OpenSSL', prefix : '''#include <openssl/ui.h>''', dependencies : libopenssl)
```

Comment:
Can this check only for the header's presence? Adding a new compilation test will slow down every build forever, for a pretty niche use case. If openssl is built with NO_UI_CONSOLE, is the header shipped at all? If not, that should be enough and should be faster

Reply:
But what's the use case for that? I thought the fix was for building against an ssl library that is built without it, no?

Reply:
But this now leaves the compilation check only, which slows down every build...

> Depends on usecase. Even OpenSSL is built with UI support, people may want to disable UI support in systemd. In that case, -DOPENSSL_NO_UI_CONSOLE=1 may be set in c_args.

But what's the use case for that? The ticket talks about other ssl implementations that don't have the feature at all, and it makes sense to support those. What's the use case for the combination with openssl which supports it?